src/Communication/Connection.php

```diff
@@ -106,6 +106,41 @@ public function __construct($socketClient, LoggerInterface $logger = null, int $
 
         // create socket client
         if (\is_string($socketClient)) {
+
+            $parsedUrl = parse_url($socketClient);
+            if (isset($parsedUrl['scheme']) && ($parsedUrl['scheme']==='http' or $parsedUrl['scheme']==='https'))
```

Comment:
The expression could be simpler:
```suggestion
            if (\in_array(\parse_url($socketClient, PHP_URL_SCHEME), ['http', 'https'])
```

Reply:
please always use strict mode, unless you have a very good reason to not do it - it's less important in modern php, but in <=7.4: https://3v4l.org/Yc3UU